Updates for new typing.

tasseff opened the pull request:
Here is a first shot at conforming to the new typing convention in InfrastructureModels packages. Let me know if this looks okay. More updates (on a separate branch) to come for greater conformity with current version of PowerModels.

ccoffrin commented on src/core/base.jl:
```diff
@@ -216,38 +216,20 @@ Some of the common keys include:
+        #    else
+        #        ref[Symbol(key)] = item
+        #    end
+        #end
```
Tests pass so you can drop these commented lines.

ccoffrin commented on src/core/data.jl:
```diff
@@ -1,3 +1,11 @@
+"Maps component types to status parameters."
+const wm_component_status = Dict("nodes" => "status", "pipes" => "status",
+                                 "pumps" => "status", "tanks" => "status")
```
You don't need this dict, becouse this code conforms to the `status` field standards.  PowerModels does not due to some legacy data model, but we are planning to resolve that in the next 6 months or so.

ccoffrin commented on src/core/data.jl:
```diff
@@ -1,3 +1,11 @@
+
+"Maps component types to inactive status values."
+const wm_component_status_inactive = Dict("nodes" => 0, "pipes" => 0,
+                                          "pumps" => 0, "tanks" => 0)
```
Same remark as the `wm_component_status` comment.

ccoffrin commented on src/core/solution.jl:
```diff
@@ -55,35 +55,44 @@ function get_solution(wm::GenericWaterModel, sol::Dict{String, <:Any})
+    #add_setpoint_pump_status!(sol, wm)
+    add_setpoint_pipe_resistance!(sol, wm)
+    add_setpoint_node_head!(sol, wm)
+    #add_setpoint_reservoir!(sol, wm)
```
Drop commented lines?

ccoffrin commented on src/core/solution.jl:
```diff
@@ -119,8 +128,250 @@ function add_pipe_resistance_setpoint(sol, wm::GenericWaterModel)
+end
+
+"adds values based on JuMP variables"
+function add_setpoint!(
```
Looking at this reminds me that we should setup some abstract solution building tools in InfrastructureModels, i.e. https://github.com/lanl-ansi/InfrastructureModels.jl/issues/1.  Feel free to remind me about next month.  :-)

ccoffrin commented on test/runtests.jl:
```diff
@@ -2,30 +2,26 @@ using WaterModels
 import Memento
 
+# Suppress warnings during testing.
+Memento.setlevel!(Memento.getlogger(InfrastructureModels), "error")
```
Looking at this makes me think we should probably add the `logger_config!` function to InfrastructureModels.

ccoffrin commented on test/runtests.jl:
```diff
@@ -2,30 +2,26 @@ using WaterModels
+
+# For checking status codes
+import MathOptInterface
+const MOI = MathOptInterface
```
You should be able to drop these in the tests if you prefer becouse WaterModels exports the MOI status codes.  This is a style choice each package can make, i.e. If the tests should emulate a user calling `using` or a package developer calling `import`.

tasseff commented:
Thanks for the comments! I've addressed your comments and think it's in better shape, now. Unless there are any objections, I'm going to merge this with master.

ccoffrin commented:
Sounds good to me.  PowerModels will be merging the type update to master early next week, but we are holding off tagging a new release until the end of Sept.
